docs/reference_guides/state_space.rst

```diff
@@ -3,156 +3,187 @@ The State Space
+    time and initial conditions.
+    The Core State Space is explicitely obtained in the module and is represented by
+    a pd.DataFrame. As such it is the basis for most calculations and every state in the
+    solution is partly represented by apointer to a row in the core state space.
 
-.. _dense_grid:
 
-Dense Grid
-----------
 
-The dense grid is a list that contains all states of dense variables. A dense variable
-is not a deterministic function of past choices and time. Adding another dense variables
-essentially copies the full state space. The disjunction makes use of this property. By
-expressing a state as a combination of a dense and a core state we avoid several
-duplications.
+.. _dense_grid:
 
+- Dense Grid:
+    The dense grid is a list that contains all states of dense variables.
+    The main difference to the core state space is that we do not apply logic to obtain
+    all feasible states of dense dimensions.
+    Instead we just use the full cartesian product of dense dimenions. As we already
+    mentioned in the last section this is either due to the fact that the marginal change
+    of the state space due to the addition of the variable is so simple that we do not
+    have to do any more than to duplicate the existing state space or due to the fact
+    the marginal change is so complicated that we are fine with having a slightly bigger
+    state space than required.
+    The dense grid is also explicitely obtained and represented by a list of tuples.
+    Each state in the solution is partly represented by a pointer to a position in the
+    dense grid.
 
 .. _dense_state_space:
 
-Dense State Space
------------------
-
-The dense state space is the full state space that contains all admissible states. respy
-does not store the full dense state space explicitly. The concept is nevertheless
-important since the model solution essentially loops through each dense state.
-
+- Dense State Space:
+    The dense state space contains all full states that the model allows for.
+    In respy the dense state space is essentially the cartesian product of the core
+    state space and the entries in the dense grid.
+    We however do not store the full dense state space explicitly.
+    We rather create the dense state space sperately for each
+    dense_period_choice_chore.
+    Each dense state is represented by a combination of the dense state
+    and the subset of the core dataframe that corresponds to the particular
+    denste_period_choice_core. We only require the full information at two
+    particular points in the creation of the state space and the solution
+    of the model. Since it takes a long time to create this object and since it
+    would consume a lot of memory to keep it in the working memory at all times
+    the objects are saved to disk after they are created and only called whenever
+    they are required.
+
+Division
```

Comment:
Partly attribute description or moved up where the separation by choice sets is explained.